sqlglot/expressions.py

```diff
@@ -731,6 +731,17 @@ class ColumnDef(Expression):
     }
 
 
+class AlterColumn(Expression):
+    arg_types = {
+        "this": True,
+        "dtype": False,
```

Comment:
why dtype and not type?

Reply:
Because we use `dtype` in other places in the codebase too for DataType objects, whereas I didn't see a mention of "type" besides the Expression slot.